pkg/controller/certificates/signer/signer.go

```diff
@@ -111,9 +112,14 @@ func NewCSRSigningController(
 
 // Run the main goroutine responsible for watching and syncing jobs.
 func (c *CSRSigningController) Run(ctx context.Context, workers int) {
-	go c.dynamicCertReloader.Run(ctx, workers)
-
-	c.certificateController.Run(ctx, workers)
+	var wg sync.WaitGroup
+	wg.Go(func() {
```

Comment:
Most of these changes seem like they ensured race conditions didn't allow us to exit without shutting down the relevant goroutine. In this case there was no apparent race condition. This controller did not try to wait for c.dynamicCertReloader.Run to exit before shutting down. Given the goal of the PR that is probably fine but flagging as this is a more fundamental change in behavior than the other changes.